Gems/Atom/RHI/Code/Source/RHI/CpuProfilerImpl.cpp

```diff
@@ -218,6 +232,18 @@ namespace AZ
             return m_enabled;
         }
 
+        const CachedTimeRegion::GroupRegionName& CpuProfilerImpl::InsertDynamicName(const char* groupName, const AZStd::string& regionName)
+        {
+            AZStd::scoped_lock lock(m_dynamicNameMutex);
+            AZ_Warning("CpuProfiler", m_dynamicRegionNameSet.size() < 16384, "Stored dynamic region names are accumulating. Consider removing a AZ_ATOM_PROFILE_DYNAMIC invocation.");
```

Comment:
Where is 16384 coming from? Maybe create a variable with a name that explains this limit.